src/utils/arfs_builders/arfs_file_builders.ts

```diff
@@ -0,0 +1,180 @@
+import {
+	ContentType,
+	deriveFileKey,
+	fileDecrypt,
+	GQLNodeInterface,
+	GQLTagInterface,
+	Utf8ArrayToStr
+} from 'ardrive-core-js';
+import Arweave from 'arweave';
+import { ArFSPrivateFile, ArFSPublicFile } from '../../arfsdao';
+import { DriveKey, FileID, TransactionID } from '../../types';
+import { ArFSFileOrFolderBuilder } from './arfs_builders';
+
+export abstract class ArFSFileBuilder<T extends ArFSPublicFile | ArFSPrivateFile> extends ArFSFileOrFolderBuilder<T> {
+	size?: number;
+	lastModifiedDate?: number;
+	dataTxId?: TransactionID;
+	dataContentType?: ContentType;
+
+	getGqlQueryParameters(): GQLTagInterface[] {
+		return [
+			{ name: 'File-Id', value: this.entityId },
+			{ name: 'Entity-Type', value: 'file' }
+		];
+	}
+}
+
+export class ArFSPublicFileBuilder extends ArFSFileBuilder<ArFSPublicFile> {
+	static fromArweaveNode(node: GQLNodeInterface, arweave: Arweave): ArFSPublicFileBuilder {
+		const { tags } = node;
+		const fileId = tags.find((tag) => tag.name === 'File-Id')?.value;
+		if (!fileId) {
+			throw new Error('File-ID tag missing!');
+		}
+		const fileBuilder = new ArFSPublicFileBuilder(fileId, arweave);
+		return fileBuilder;
+	}
+
+	protected async buildEntity(): Promise<ArFSPublicFile> {
+		if (
+			this.appName?.length &&
+			this.appVersion?.length &&
+			this.arFS?.length &&
+			this.contentType?.length &&
+			this.driveId?.length &&
+			this.entityType?.length &&
+			this.txId?.length &&
+			this.unixTime &&
+			this.parentFolderId?.length &&
+			this.entityId?.length
+		) {
+			const txData = await this.arweave.transactions.getData(this.txId, { decode: true });
+			const dataString = await Utf8ArrayToStr(txData);
+			const dataJSON = await JSON.parse(dataString);
+
+			// Get the file name
+			this.name = dataJSON.name;
+			if (!this.name) {
+				throw new Error('Invalid file state');
+			}
+
+			// TODO: WHY IS THE ArFSPublicFile type missing these fields?
```

Comment:
We should make these available to the user. These fields relevant and useful to the user. And when we go to implement move-file, having these not available on the meta data makes it more difficult to rebuild the transaction.